src/Solver/time_stepping/TimeManager.h

```diff
@@ -186,6 +186,10 @@ class seissol::time_stepping::TimeManager {
     void printComputationTime(const std::string& outputPrefix, bool isLoopStatisticsNetcdfOutputOn);
 
     void freeDynamicResources();
+
+    inline TimeStepping* getTimeStepping() {
```

Comment:
Make reference. Or maybe also const? If there's only one parameter you want to modify, it may look more elegant to offer a method to modify it directly.

Reply:
This was done by design for `initializeCellLocalMatrices` and similar methods. As far as I can recall, it was not done for the ITM. We just used the `initalizeCellLocalMatrices` to do what we want to do.

Reply:
Thanks, I see now—it's because you want to have a reference in the ITM manager. The `initalizeCellLocalMatrices` does not use a pointer, but a reference. It's only to store the reference for later usage that we need the pointer here.

You _could_ also pass a reference from the `TimeManager` which is then converted to a pointer in the ITM manager, but that would just move the problem. Quite honestly, it's not ideal to have it that way exposing a pointer, but I don't see much possibility to change it unless we want to refactor the basic structure of Seissol a bit (which is arguably out of scope for this PR).

Can you maybe make the pointer `const`?

Reply:
Making the pointer `const` makes the compiler complain of invalid conversion once again. I think it is better to let it stay untouched as it is being used by other methods too

Reply:
Did you also make it const in the `initializeTimeMirrorManagers` and the ITM manager classes themselves—and their parameters? The conversion from const ptr to const ref should work

Reply:
Ach! Yes, that makes more sense. Thank you! Fixed it now.